frame/support/src/traits/tokens/nonfungibles_v2.rs

```diff
@@ -27,19 +27,22 @@
 //! Implementations of these traits may be converted to implementations of corresponding
 //! `nonfungible` traits by using the `nonfungible::ItemOf` type adapter.
 
-use crate::dispatch::{DispatchError, DispatchResult};
+use crate::{
+	dispatch::{DispatchError, DispatchResult},
+	traits::tokens::{CollectionId, DestroyWitness, ItemId},
+};
 use codec::{Decode, Encode};
 use sp_runtime::TokenError;
 use sp_std::prelude::*;
 
 /// Trait for providing an interface to many read-only NFT-like sets of items.
 pub trait Inspect<AccountId> {
 	/// Type for identifying an item.
-	type ItemId;
+	type ItemId: ItemId;
```

Comment:
```suggestion
	type ItemId: Parameter;
```

Reply:
Maybe also `+ Copy`, but not sure we should have this here.

Reply:
@jsidorenko what do you think?